packages/console-types/src/types/project.ts

```diff
@@ -217,6 +218,10 @@ export const strategySchema = {
 export const environment = {
   type: 'object',
   properties: {
+    type: {
+      type: 'string',
+      enum: Object.values(ENVIRONMENT_TYPES),
```

Comment:
doesn't this make the object non constant? I'm not sure the FromSchema works well with this runtime evaluation

Reply:
`Object.values` correctly infers the type `"runtime" | "virtual" | undefined`